### DISCUSSION TITLE
Kratos on kubernetes / k8s / Istio Mesh - upstream connect error

### DISCUSSION THREAD

**Discussion by senitsgh** [ory/kratos#2239] [Q&A] [answered]
Hi 
I have kratos running on kubernetes/istio service mesh. I have the UI Server and Kratos Server on the same domain.
All self service flows work fine, except for the sign up. Where the user is getting registered successfully, yet the Browser ends up with `upstream connect error or disconnect/reset before headers. reset reason: connection termination`. 
 
<img width="934" alt="image" src="https://user-images.githubusercontent.com/47174355/154817430-a8b40af6-3b4d-416f-955e-44c4d029fcc0.png">

SignUp flow :- self-service/registration?flow=312dc8b6-b987-4ea3-b560-9c14c915d24c

I have been trying to solve this for the past few days with no luck. 
Has anyone come across such issue with Kratos? Any help would be greatly appreciated :)

**Comment by senitsgh** [comment, marked as answer]
Found the root cause and adding the solution here if anyone in future faces this issue.
Kratos calls an external service at (api.pwnedpasswords.com) to validate compromised passwords. Since my service mesh/k8s was not allowing the outbound call to this service, the requests were failing with upstream connect error. 

I had to add a 'ServiceEntry' for 'api.pwnedpasswords.com' to make it work :)

**Reply by aeneasr** [reply]
Thank you for following up! It is now also possible to disable this check in the configuration, but of course we do not recommend to skip it if possible :)

**Comment by adamstrawson** [comment]
@senitsgh Thank you for adding this GitHub issue, I've been trying to find the cause of Kratos timeouts for a few days, and your issue led me to the cause! ❤️
